### PR TITLE
Use `pytest_configure` to dynamically adding command line options 

### LINKED ISSUE #10027: Replace `pytest` with `dev/pytest.sh` in `mlflow/ml-package-versions.yml`
### Summary

A follow-up task for https://github.com/mlflow/mlflow/pull/10006. Replace `pytest` with `dev/pytest.sh` in `mlflow/ml-package-versions.yml` to enable the `fail-fast` behavior.

Example:

```diff
diff --git a/mlflow/ml-package-versions.yml b/mlflow/ml-package-versions.yml
index 4fdc2635d..373aa9e17 100644
--- a/mlflow/ml-package-versions.yml
+++ b/mlflow/ml-package-versions.yml
@@ -229,7 +229,7 @@ fastai:
       "< 2.8.0": ["torch<1.13.0", "torchvision<0.14.0"]
       ">= 2.8.0": ["torch", "torchvision"]
     run: |
-      pytest tests/fastai/test_fastai_autolog.py
+      dev/pytest.sh tests/fastai/test_fastai_autolog.py
 
 onnx:
   package_info:
```` 

### Notes

- Make sure to open a PR from a **non-master** branch.
- Sign off the commit using the `-s` flag when making a commit:

  ```sh
  git commit -s -m "..."
           # ^^ make sure to use this
  ```

- Include `Resolve #{issue_number}` (e.g. `Resolve #123`) in the PR description when opening a PR.

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/10361?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10361/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10361
```

</p>
</details>

### Related Issues/PRs

Resolve #10027

### What changes are proposed in this pull request?

Use `pytest_configure` to dynamically set the `maxfail` option. This allows us to remove `dev/pytest.sh` and simplify the github actions configuration files.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/sklearn/test_sklearn_model_export.py [hunk 1]
```diff
@@ -121,6 +121,7 @@ def sklearn_custom_env(tmp_path):
 
 
 def test_model_save_load(sklearn_knn_model, model_path):
+    assert False
```
Comment (harupy) [inline tests/sklearn/test_sklearn_model_export.py:124]: test, will revert

Comment (harupy) [reply tests/sklearn/test_sklearn_model_export.py:124]: worked as expected: https://github.com/mlflow/mlflow/actions/runs/6836235927/job/18590929638?pr=10361

Comment (serena-ruan) [reply tests/sklearn/test_sklearn_model_export.py:124]: Revert this?

**Review by serena-ruan — APPROVED** [review]
LGTM once the test is reverted

File: tests/sklearn/test_sklearn_model_export.py (hunk 1 above)
Comment (harupy) [inline tests/sklearn/test_sklearn_model_export.py:124]:
```suggestion
```

Comment (harupy) [reply tests/sklearn/test_sklearn_model_export.py]: reverted

**Review by B-Step62 — APPROVED** [review]
lgtm!

**Pull request merged**